docs: add integration guide for external prefill/decode workloads

googs1025 opened the pull request:
Add integration guide for external prefill/decode workload.
Include instructions on using `by-label` plugin in `EndpointPickerConfig`
to support custom labeling schemes (e.g., role=prefill/decode) from external systems.  

Fix https://github.com/llm-d/llm-d-inference-scheduler/issues/436 and https://github.com/llm-d/llm-d-inference-scheduler/issues/430

elevran commented on docs/dp.md:
```diff
@@ -119,6 +119,77 @@ This evolved version removes the requirement for sidecars on the **prefill node*
+
+The llm-d inference scheduler supports integration with external disaggregated prefill/decode (P/D) workloads other inference frameworks—that follow the same P/D separation pattern but use **different Kubernetes Pod labeling conventions**.
+
+#### Labeling Convention Flexibility
```
Is the jump by 2`#` from `##` to `####` intentional?

googs1025 commented on docs/dp.md:
```diff
@@ -119,6 +119,77 @@ This evolved version removes the requirement for sidecars on the **prefill node*
+plugins:
+  # Prefill selection: match Pods with label role=prefill
+  - type: by-label
+    name: "external-prefill"
```
done

googs1025 commented:
> > Thanks for the thorough review! I’ve addressed all the nits. And I renamed the file to disagg-pd.md for clarity.
> 
> There are files in the repo that refer to the old name (`dp.md`) - failing markdown link check test. Please either revert or fix references to use the new file name. Grepping for "md.dp" returned `docs/architecture.md` and `README.md`.

thanks for reminding this. 😄  fixed
